### ISSUE TITLE
Relation between comparison model and softregex

### ISSUE THREAD

**Issue by abdulazizab2** [jacger2/softregex#1]
Hello,
I have read your paper and it is really interesting. I am trying to understand what is the role of the comparison model in generating the regex from natural language. As far as I understand in comparison mode, you are training to capture if two regex give similar results or not. Later, in the softregex model you make a seq2seq model to generate a regex. What is the role of the comparison model in softregex, is it used as a loss function?

**Comment by jacger2** [comment]
Thank you for having interest in our research. Yes, the comparison model is used in a loss function. As you can see in the softregex paper, two different regular expressions can have semantically equal meaning. However training method (MLE base) of a basic seq2seq model trains each natural language sentence with a single correct answer(regex). Therefore we additionally trains nl-regex model (policy gradient) to teach other regular expressions, which has different syntatic but same meaning of ground-truth regex, using the regex companion model.
